### ISSUE TITLE
FEN generation doesn't handle en passant

### ISSUE THREAD

**Issue by davidssmith** [niklasf/python-chess#66]
After 1. e4 e5, the FEN generated is 
```
rnbqkbnr/pppp1ppp/8/4p3/4P3/8/PPPP1PPP/RNBQKBNR w KQkq - 0 2
```
ignoring the en passant square. According to the [FEN definition](https://en.wikipedia.org/wiki/Forsyth%E2%80%93Edwards_Notation), the correct FEN should be 
```
rnbqkbnr/pppp1ppp/8/4p3/4P3/8/PPPP1PPP/RNBQKBNR w KQkq e6 0 2
```

The en passant square is supposed to be included whether or not a pawn in is position to capture en passant. Should be as simple as testing for a two-square pawn move and then setting the en passant square behind it.

**Comment by niklasf** [comment]
Hi.

There is a story behind this. I'll summarize the main point:

It was a conscious decision to only display an en-passant square, when there is a legal en-passant move. This even goes a step further than [X-FEN](https://en.wikipedia.org/wiki/X-FEN), which has en-passant squares only when there is a pseudo-legal en-passant move.

I usually cite this argument, which I picked up from talkchess: "If one thinks that `rnbqkbnr/pppp1ppp/8/4p3/4P3/8/PPPP1PPP/RNBQKBNR w KQkq - 0 2` and `rnbqkbnr/pppp1ppp/8/4p3/4P3/8/PPPP1PPP/RNBQKBNR w KQkq e6 0 2` are different positions, then they are playing a different game."

As an avid supporter of doing it this way, I'd be intrested in any software that is incomptable with it or real-world scenarios, where it is problematic.

If you absolutely need it (untested):

```python
def oldstyle_fen(board):
    builder = []
    builder.append(board.board_fen())
    builder.append("w" if board.turn == chess.WHITE else "b")
    builder.append(board.castling_xfen())
    builder.append(chess.SQUARE_NAMES[board.ep_square] if board.ep_square else "-")
    builder.append(str(board.halfmove_clock))
    builder.append(str(board.fullmove_number))
    return " ".join(builder)
```

**Comment by davidssmith** [comment]
You make a good case. Although having the e.p. square makes your FENs a superset of those that don't. A program that ignores it will be compatible with FENs that include it, while a program that expects it will have problems if it is absent.  Depends on your goals I guess.

As an example, this behavior is incompatible with chessx, which does add the e.p. square. I found that when I generated FENs with chessx, I couldn't use them directly with your package if there is an e.p. square.

Thanks for the code snippet, and thanks for a great package. You saved me a ton of work writing my own PGN parser.
